apps/system/js/bluetooth.js

```diff
@@ -216,6 +221,67 @@ var Bluetooth = {
   },
 
   /**
+   * Get adapter from bluetooth through promise interface.
+   * Since BTv1 get onenabled event before onadapteradded event,
+   * We need watch if adapter is retrieved.
+   *
+   * @public
+   * @return {Promise} Bluetooth Adapter
+   */
+  adapter: function bt__adapter() {
+    return new Promise((resolve) => {
+      if (this.defaultAdapter !== null) {
+        resolve(this.defaultAdapter);
+      }
+
+      this.watch('defaultAdapter', (id, oldval, newval) => {
+        if (newval !== null) {
+          this.unwatch('defaultAdapter');
+          resolve(newval);
+          return newval;
+        }
+      });
+    });
+  },
+
+  /**
+   * Return device pairing result.
+   *
+   * @public
+   * @param {string} mac target device address
+   * @return {Promise} paired result
+   */
+  pair: function bt__pair(mac) {
+    return new Promise((resolve, reject) => {
+      var req = this._adapter.pair(mac);
+      req.onsuccess = () => {
+        resolve();
+      };
+      req.onerror = () => {
+        reject();
```

Comment:
Since we `reject` the promise in `onerror` function, It's better to pass the error reason via `reject(reason)` here. The reason should be `req.error.name`.